### ISSUE TITLE
request headers are not being normalized and overrides not respected

### ISSUE THREAD

**Issue by knksmith57** [tomas/needle#167]
Currently, when a [request is setup](https://github.com/tomas/needle/blob/master/lib/needle.js#L252) and the [default](https://github.com/tomas/needle/blob/master/lib/needle.js#L187) [headers](https://github.com/tomas/needle/blob/master/lib/needle.js#L195) are [merged into the request-specific ones](https://github.com/tomas/needle/blob/master/lib/needle.js#L202), no normalization is being performed.

This is problematic because, per [RFC2616](https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2), request headers are __case-insensitive__. The result is that header overrides are not respected and duplicate headers with the same field name (of mixed casing) but different values could be sent in the request.

For example:

```
needle.get('http://example.com', {headers: {'user-agent': 'not going to work'}})
```

will result in the following `config.headers` object:
```
{
  Accept: '*/*,
  User-Agent: ''Needle/x.x.x (Node.js vx.x.x; <platform> <arch>)",
  user-agent: "not going to work"
}
```

where the expected object would be:
```
{
  Accept: '*/*,
  user-agent: "not going to work"
}
```

I think the simplest solution would be to do the following:

1. move the [header overrides](https://github.com/tomas/needle/blob/master/lib/needle.js#L201) down below the [basic/digest auth section](https://github.com/tomas/needle/blob/master/lib/needle.js#L205) (ie: to ~ [line 235](https://github.com/tomas/needle/blob/master/lib/needle.js#L235)) and ensure the header key is lowercased:

          for (var h in options.headers)
            config.headers[h.toLowerCase()] = options.headers[h];

2. add the following directly above the overrides:

          for (var h in config.headers)
            config.headers[h] = String(config.headers[h]).toLowerCase();

This tracks the current time complexity and ensures overrides are respected.

**Comment by tomas** [comment]
Good point. I'll take a look at this now.

**Comment by knksmith57** [comment]
@tomas hey sir, not sure if you had a chance to take a look at this yesterday. Did you want me to work a PR right quick? Happy to help!

**Comment by tomas** [comment]
Yes! A PR would certainly help. I'm out of town at the moment so much slower than usual. :)

**Comment by tomas** [comment]
This just landed in d227ef4. I ended up changing all header logic to lowercase!

Thanks again mate.
